x-pack/plugins/ml/public/embeddables/anomaly_charts/use_anomaly_charts_input_resolver.ts

```diff
@@ -0,0 +1,219 @@
+    { data: dataServices },
+    { anomalyDetectorService, anomalyExplorerService, mlResultsService },
+  ] = services;
+  const { timefilter } = dataServices.query.timefilter;
+
+  const [chartsData, setChartsData] = useState<any>();
+  const [error, setError] = useState<Error | null>();
+  const [isLoading, setIsLoading] = useState(false);
+
+  const chartWidth$ = useMemo(() => new Subject<number>(), []);
+  const severity$ = useMemo(() => new Subject<number>(), []);
+
+  const timeBuckets = useMemo(() => {
+    return new TimeBuckets({
+      'histogram:maxBars': uiSettings.get(UI_SETTINGS.HISTOGRAM_MAX_BARS),
+      'histogram:barTarget': uiSettings.get(UI_SETTINGS.HISTOGRAM_BAR_TARGET),
+      dateFormat: uiSettings.get('dateFormat'),
+      'dateFormat:scaled': uiSettings.get('dateFormat:scaled'),
+    });
+  }, []);
+
+  useEffect(() => {
+    const subscription = combineLatest([
+      getJobsObservable(embeddableInput, anomalyDetectorService, setError),
+      embeddableInput,
+      chartWidth$.pipe(skipWhile((v) => !v)),
+      severity$,
+      refresh.pipe(startWith(null)),
+    ])
+      .pipe(
+        tap(setIsLoading.bind(null, true)),
+        debounceTime(FETCH_RESULTS_DEBOUNCE_MS),
+        switchMap(([jobs, input, embeddableContainerWidth, severityValue]) => {
+          if (!jobs) {
+            // couldn't load the list of jobs
+            return of(undefined);
+          }
+
+          const { maxSeriesToPlot, timeRange: timeRangeInput, filters, query } = input;
+
+          const viewBySwimlaneFieldName = OVERALL_LABEL;
+
+          anomalyExplorerService.setTimeRange(timeRangeInput);
+
+          const explorerJobs: ExplorerJob[] = jobs.map((job) => {
+            const bucketSpan = parseInterval(job.analysis_config.bucket_span);
+            return {
+              id: job.job_id,
+              selected: true,
+              bucketSpanSeconds: bucketSpan!.asSeconds(),
+            };
+          });
+
+          let influencersFilterQuery: InfluencersFilterQuery;
+          try {
+            influencersFilterQuery = processFilters(filters, query);
+          } catch (e) {
+            // handle query syntax errors
+            setError(e);
+            return of(undefined);
+          }
+
+          const bounds = anomalyExplorerService.getTimeBounds();
+
+          // Can be from input time range or from the timefilter bar
+          const selections: AppStateSelectedCells = {
+            lanes: [OVERALL_LABEL],
+            times: [bounds.min?.unix()!, bounds.max?.unix()!],
+            type: SWIMLANE_TYPE.OVERALL,
+          };
+
+          const selectionInfluencers = getSelectionInfluencers(selections, viewBySwimlaneFieldName);
+
+          const jobIds = getSelectionJobIds(selections, explorerJobs);
+
+          const bucketInterval = timeBuckets.getInterval();
+
+          const timeRange = getSelectionTimeRange(selections, bucketInterval.asSeconds(), bounds);
+          const explorer$ = forkJoin({
```

Comment:
variable is redundant 
```suggestion
          return forkJoin({
```

Reply:
Updated here [`14d383b` (#94396)](https://github.com/elastic/kibana/pull/94396/commits/14d383b9543d14cc423ae77ebd3b5ee87f817d8e)